python/lsst/cp/pipe/utilsEfd.py

```diff
@@ -257,41 +319,164 @@ def parseMonochromatorStatus(self, data, dateStr):
+        outResults = outResults[outResults["level"] == 20]
+
+        # These will be new columns
+        intensityMean = []
+        intensityStdev = []
+        intensityTime = []
+        intensityFile = []
+
+        for row in outResults:
+            # Fallback values in case the regexp fails
+            mean = np.nan
+            stdev = np.nan
+            time_mean = np.nan
+            filename = "REGEXP_FAIL"
+
+            # Find the last "filename\.extension" before the first newline;
+            #      the last [] grouped values before the second newline;
+            #      the last [] grouped values before the end of the string.
+            magic = re.findall(r"\b\w+.+?(\w+?\.\w+)\n\b\w+.+\[(.*?)\]\n\b\w+.+\[(.*?)\]$",
+                               row["message"])
+            if len(magic) != 0:
+                # If we matched, split the grouped values, cast them to floats.
+                filename, intensity_str, time_str = magic[0]
+                mean, median, stdev = intensity_str.split(",")
+                time_mean, time_median = time_str.split(",")
+                mean = float(mean)
+                median = float(median)
+                stdev = float(stdev)
+                time_mean = float(time_mean)
+                time_median = float(time_median)
+                # Censor the saturated points so plots look nice.
+                if np.abs(mean) > 1e37:
+                    mean = np.nan
+
+            intensityMean.append(mean)
+            intensityStdev.append(stdev)
+            intensityFile.append(filename)
+            intensityTime.append(time_mean)
+
+        # Add our new columns at the start of the column list
+        outResults.add_column(intensityMean, name="intensity", index=1)
+        outResults.add_column(intensityStdev, name="intensityStd", index=2)
+        outResults.add_column(intensityTime, name="intensityTimeMean", index=3)
+        outResults.add_column(intensityFile, name="expectedLfaFile", index=4)
+        return outResults
+
+    def parseElectrometerStatus(self, data, dateStr, dateEnd=None,
+                                doIntegrateSamples=False, index=201):
+        """Determine electrometer status for a specific date.
+
+        Parameters
+        ----------
+        data : `astropy.table.Table`
+            The dataframe of monochromator results from the EFD.
+        dateStr : `str`
+            The date to look up in the status for.
+        dateEnd : `str`
+            The end date to look in the status for.
+        doIntegrateSamples: `bool`
+            If true, take the average of all samples between
+            ``dateStr`` and ``dateEnd``.
+        index : `int`
+            The salIndex of the device we want to read.  For LATISS,
+            this should be 201.  For the main telescope, 101.
+
+        Returns
+        -------
+        indexDate : `str`
+            Date string indicating the monochromator state change.
```

Comment:
Is this utc or tai in the return?

Reply:
As per previous comment: Everything is now in TAI.